torch/_dynamo/variables/builder.py

```diff
@@ -700,7 +711,12 @@ def wrap_literal(self, value):
             )
 
     def wrap_tensor(self, value: torch.Tensor):
-        if self.get_source().guard_source().is_nn_module():
+        source = self.get_source()
+
+        if (
+            source.guard_source().is_nn_module()
+            and not source.guard_source().is_fsdp_module()
```

Comment:
Why are you skipping registration? This is not correct and will result in missing FQNs.

Reply:
Also, if you get rid of this, you can get rid of alllll the `is_fsdp_module` logic and just do what https://github.com/pytorch/pytorch/pull/96905 does. Or did you find that PR missing some cases?